src/Microsoft.SqlTools.ServiceLayer/AutoParameterizaition/SqlParameterizer.cs

```diff
@@ -0,0 +1,128 @@
+﻿//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.SqlServer.TransactSql.ScriptDom;
+using System.Linq;
+using Microsoft.SqlTools.ServiceLayer.AutoParameterizaition.Exceptions;
+using System.Data.Common;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.AutoParameterizaition
+{
+    public static class SqlParameterizer
```

Comment:
So I assume a lot of this logic is copied from the SSMS implementation? Have you looked into what it would take to have this stuff be in a common module that both use instead of having all this duplicated code?

Reply:
There are some differences. For example in SSMS, this class is not static nor it the parameterization an extension method on the DbCommand object. Also, the code sense returns an ADS specific ScriptFileMarker instead of whatever SSMS uses for the UI to display information. Other than that, the core is basically the same stuff. I will take a look and see what it will take to extract and centralize the similar logic. I am sure we could adapt to the differences.